### ISSUE TITLE
feature request: save screenshot of a window

### ISSUE THREAD

**Issue by gaika** [arrayfire/arrayfire#1721]
May be just a way to access current window pixels as AFArray and then `save_image` it?

**Comment by pavanky** [comment]
@gaika programatically or from the UI ?

**Comment by gaika** [comment]
programmatically

**Comment by pavanky** [comment]
@9prady9 Does this need to be implemented in forge or can this be done in arrayfire?

**Comment by gaika** [comment]
I think it should be possible in arrayfire with `glReadPixels`

**Comment by shehzan10** [comment]
Forge has API for this https://github.com/arrayfire/forge/blob/devel/include/fg/window.h#L521. But I don't think there is a corresponding API call in ArrayFire.

**Comment by pavanky** [comment]
@shehzan10 That function doesn't take in a format. Is it figuring it out based on the extension ?

**Comment by shehzan10** [comment]
Yeah, it's similar to ArrayFire's saveImage, just that it's saving the framebuffer.

So the only thing needed to do is to is to create API in ArrayFire that would call the forge function. Should be simple enough for an external contribution.
